analytics/loggable_objects.go

```diff
@@ -0,0 +1,96 @@
+package analytics
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"net/http"
+)
+
+type RequestType string
+
+const (
+	COOKIE_SYNC RequestType = "cookie_sync"
+	AUCTION     RequestType = "auction"
+	SETUID      RequestType = "set_uid"
+	AMP         RequestType = "amp"
+)
+
+//Loggable object of a transaction at /openrtb2/auction endpoint
+type AuctionObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   openrtb.BidRequest
+	Response  openrtb.BidResponse
+	UserAgent string
+}
+
+type AmpObject struct {
+	Type      RequestType
+	Status    int
+	Error     []error
+	Request   string
+	Response  string
+	UserAgent string
+	Origin    string
+}
+
+//Bid requests from each adapter. This information is available in /openrtb2/auction response but necessary to log in case there are errors
+type LoggableAdapterRequest struct {
+	Name     string
+	Requests string
+	Uri      string
+	Header   http.Header
+	Method   string
+}
+
+//Loggable object of a transaction at /setuid
+type SetUIDObject struct {
+	Type    RequestType
+	Status  int
+	Error   []error
+	Cookie  string
+	Success bool
+}
+
+//Loggable object of a transaction at /cookie_sync
+type CookieSyncObject struct {
+	Type     RequestType
+	Status   int
+	Error    []error
+	Request  string
+	Response string
```

Comment:
This looks like it can be structured too. Important pieces of data (IMO) are:

- which bidders did the request include?
- which bidders already had syncs?